### PR TITLE
e2e docs

### PR DESCRIPTION
**Pull request by vitorflg**
As the tests are not ready yet, the docs will still need many updates but it's a starting point :)

### CODE REVIEW

File: docs/e2e-testing.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+## Overview
+
+[Cypress](https://github.com/cypress-io/cypress) is a testing framework that provide a great experience to write end-to-end (E2E) tests for modern web applications. It's very flexible, fast, easy to use, and besides being broadly used in the javascript ecosystem, it works very well with Gatsby, that's why we choose it as our main tool.
```
Comment (gris) [inline docs/e2e-testing.md:3]:
```suggestion
[Cypress](https://github.com/cypress-io/cypress) is a testing framework that provides a great experience to write end-to-end (E2E) tests for modern web applications. It's very flexible, fast, easy to use, and besides being broadly used in the JavaScript ecosystem, it works very well with Gatsby, that's why we choose it as our main tool.
```

File: docs/e2e-testing.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+Feel free to check some other alternatives and compare the trade-offs of end-to-end test tools [here](https://stackshare.io/cypress/alternatives).
```
Comment (gris) [inline docs/e2e-testing.md:5]: Do you think that is necessary?

**Review by gris — CHANGES_REQUESTED** [review]

File: docs/e2e-testing.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+   		cy.get("input[aria-label="Search"]").type("Eyeglasses")
+
+   		cy.contains('li', '[PRODUCT_NAME]').click()
+
+   		cy.contains('h1', '[PRODUCT_NAME')
+   	})
+   })
+
+   ```
+
```
Comment (carolinamenezes) [inline docs/e2e-testing.md:71]:
```suggestion

>ℹ️ *This test checks how our SFJ store behaves when it receives "Eyeglasses" as an input in our search bar.*

>ℹ️ *To build your own tests, we strongly encourage you to check Cypress's [guides](https://docs.cypress.io/guides/getting-started/writing-your-first-test.html#Add-a-test-file), [examples](https://docs.cypress.io/examples/examples/recipes.html#Fundamentals), [API reference](https://docs.cypress.io/api/api/table-of-contents.html), and [best practices](https://docs.cypress.io/guides/references/best-practices.html).*

```

File: docs/e2e-testing.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+3. Run `npm run test:e2e` and verify if all tests are passing.
```
Comment (carolinamenezes) [inline docs/e2e-testing.md:72]:
```suggestion
4. Run `npm run test:e2e` to check the status of the tests.
```

File: docs/e2e-testing.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+[Cypress Documentation](https://docs.cypress.io/guides/overview/why-cypress.html#In-a-nutshell)
+
+[Cypress API](https://docs.cypress.io/api/api/table-of-contents.html)
+
+[Cypress Examples](https://docs.cypress.io/examples/examples/recipes.html)
+
+[VTEX Examples](https://github.com/vtex-sites/storecomponents.store)
+
+[Best Practices](https://docs.cypress.io/guides/references/best-practices.html)
```
Comment (carolinamenezes) [inline docs/e2e-testing.md:96]:
```suggestion
```

File: docs/e2e-testing.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+3. Run `npm run test:e2e` and verify if all tests are passing.
+
+## Code Coverage
+
+To visualize the project code coverage just run `npm run coverage`... (WIP)
+
+## Hooks
+
+VTEX has a hub of helper functions to make writing E2E tests easy, you can install .. (WIP)
```
Comment (carolinamenezes) [inline docs/e2e-testing.md:80]:
```suggestion

That's it! Additionally, you can run `npm run coverage` (WIP) to visualize the test's code coverage.

```

File: docs/e2e-testing.md (hunk 2 above)
Comment (vitorflg) [reply docs/e2e-testing.md:5]: No, we can remove it!

File: docs/e2e-testing.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+## Getting Started
+
+1. Install the `cypress` and the `start-server-and-test`(permit using Gatsby’s development server with Cypress).
+
+   ```bash
+     $ yarn add cypress start-server-and-test --dev
+   ```
+
+2. Add theses scripts into your `package.json`. `cy:open` will open the cypress [dashboard](https://www.cypress.io/dashboard) and `test:e2e` will run all the tests in the command line.
```
[the remark this answers is not in the document]
Comment (vitorflg) [reply docs/e2e-testing.md:22]: Can we keep the explanations somewhere?

**Comment by vitorflg** [comment]
Thanks a lot, @carolinamenezes !!

File: docs/e2e-testing.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (carolinamenezes) [reply docs/e2e-testing.md:22]: @vitorflg Coloquei depois do código como um callout. Você acha melhor colocar antes?

**Review by carolinamenezes — APPROVED** [review]

File: docs/e2e-testing.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+### Step 2: Writing a sample test
+
+In the following, we'll create a simple test that verifies how a specific product search behaves. Use it as an example to create your own E2E test with Cypress.
+
+1. Open up your SFJ project in the code editor of your choosing.
+2. Create a new file named `index.js` inside `cypress/e2e/{test-name}` (e.g., `cypress/e2e/search`).
+3. Using JavaScript, write your E2E test. Take the code below as an example:
+
+   ```bash
```
Comment (gris) [inline docs/e2e-testing.md:67]:
```suggestion
   ```js
```

**Pull request merged**